### ISSUE TITLE
Game runs out of Memory on Linux with 16GB of RAM

### ISSUE THREAD

**Issue by JanThomas118** [StrangeLoopGames/EcoIssues#18448]
I have just bought the game on steam to try out the linux support,
My PC:
- OS: Fedora 32
- GPU: Nvidia 1050ti with latest proprietary driver
- RAM: 16gb
- CPU: Ryzen 1600

I have set the shadow quality to simple and disabled all graphical effects on the right pane.

 The OS uses around 1.5GB when the game is not running

When i start the game in singleplayer mode the game freezes as soon as you join the world. During this it consumes around 14GB of ram.

This is less bad when joining other servers, then the game "only" uses 11GB but when you leave a server and join another one the game will crash again.

Log: https://pastebin.com/NRLpMWz6

My roommate wanted to buy this game too, but i assume this wouldn't get any better when he only has 12 gigs of ram.

**Comment by sashapony** [comment]
I have the same issue where the game goes from about 2gb on the main menu and then skyrockets to 10+ loading into a brand new world. I am using Manjaro Linux

CPU AMD Ryzen 9 3900x
GPU nvidia RTX 2070 with latest proprietary Driver
Ram 32 gb

**Comment by JanThomas118** [comment]
I have tried this again with the "Steam Linux Runtime - Soldier" which is basically a pseudo linux distro steam games should work with. And the issues persist. So this would make me think the issue has nothing to do with distro specifics

**Comment by JanThomas118** [comment]
I have tried this on my roommates PC aswell. 
- Fedora 32
- AMD RX480 (amdgpu linux driver)
- RAM 12GB
- CPU i5 4690K

The same issue is happening, the game uses 11Gigabytes of total ram until the OS kills it due to low memory

**Comment by mirasrael** [comment]
Seems related to Minimap shader issues (https://github.com/StrangeLoopGames/EcoIssues/issues/18413).
```
OutOfMemoryException: Out of memory
  at (wrapper managed-to-native) System.Object.__icall_wrapper_ves_icall_array_new_specific(intptr,int)
  at System.Collections.Generic.List`1[T].set_Capacity (System.Int32 value) [0x00021] in <4756199cf52a4f14b33cdcc5659f782e>:0 
  at System.Collections.Generic.List`1[T].EnsureCapacity (System.Int32 min) [0x00036] in <4756199cf52a4f14b33cdcc5659f782e>:0 
  at System.Collections.Generic.List`1[T].Add (T item) [0x00010] in <4756199cf52a4f14b33cdcc5659f782e>:0 
  at MinimapRigMesh.GeneratePatchesMesh (Eco.Shared.Math.Vector2i worldSize, System.UInt32 patchSize) [0x0006f] in <64f97b0e1a0845f794227ffd6cc86fde>:0 
  at MinimapRigMesh.Rewrap () [0x0005e] in <64f97b0e1a0845f794227ffd6cc86fde>:0 
  at MinimapRigMesh.Initialize () [0x00084] in <64f97b0e1a0845f794227ffd6cc86fde>:0 
  at MinimapRigMesh.Start () [0x00000] in <64f97b0e1a0845f794227ffd6cc86fde>:0 
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
EcoEngine.Logging.LogManager:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
```
It tried to build a mesh for tessellation level 1 (which is huge).

**Comment by SlayksWood** [comment]
@JanThomas118 should be fixed in next update(9.0.5). Reopen please if you still have any problem
